sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -743,6 +743,41 @@ def test_precompute_invalid_argument():
                         "Got 'auto'", ElasticNet(precompute='auto').fit, X, y)
 
 
+def test_precompute_incorrect_gram():
+    X, y, _, _ = build_dataset()
+
+    rng = np.random.RandomState(0)
+
+    X_centered = X - np.average(X, axis=0)
+    garbage = rng.standard_normal(X.shape)
+    precompute = np.dot(garbage.T, garbage)
+
+    clf = ElasticNet(alpha=0.01, precompute=precompute)
+
+    assert_raises_regex(ValueError, "Gram matrix.*did not pass validation.*",
+                        clf.fit, X_centered, y)
+
+
+def test_precompute_gram_weighted_samples():
+    X, y, _, _ = build_dataset()
+
+    rng = np.random.RandomState(0)
+    sample_weight = rng.lognormal(size=y.shape)
+
+    w_norm = sample_weight * (y.shape / np.sum(sample_weight))
+    X_c = (X - np.average(X, axis=0, weights=w_norm))
+    X_r = X_c * np.sqrt(w_norm)[:, np.newaxis]
+    Gram = np.dot(X_r.T, X_r)
+
+    clf1 = ElasticNet(alpha=0.01, precompute=Gram)
+    clf1.fit(X_c, y, sample_weight=sample_weight.copy())
```

Comment:
Uhm, could you remove the copy. It should not have any effect because we are not supposed to change it in place.
If this is the case, we have another underlying bug.

Reply:
yes, this appears to be a bug. I will file a separate issue for it.

Reply:
I have filed #19044 for this issue, if it is OK, I will add a TODO to the test and remove the copy when the bug is resolved (which I'm happy to do after this PR).